src/main/java/fi/otavanopisto/restfulptv/server/services/ServiceEntityUpdater.java

```diff
@@ -95,25 +101,35 @@ public void onServiceIdUpdateRequest(@Observes ServiceIdUpdateRequest event) {
       }
     }
   }
-  
+
   @Timeout
   public void timeout(Timer timer) {
     if (!stopped) {
-      if (!queue.isEmpty()) {
-        String entityId = queue.iterator().next();
-        if (!queue.remove(entityId)) {
-          logger.warning(String.format("Could not remove %s from queue", entityId));
-        }
-        
-        ApiResponse<IVmOpenApiService> response = ptvApi.getServiceApi().apiServiceByIdGet(entityId);
-        if (response.isOk()) {
-          cacheResponse(entityId, response.getResponse());
-        } else {
-          logger.warning(String.format("Service %s caching failed on [%d] %s", entityId, response.getStatus(), response.getMessage()));
+      if (running) {
+        return;
+      }
+
+      try {
+        running = true;
+        if (!queue.isEmpty()) {
+          String entityId = queue.iterator().next();
+          if (!queue.remove(entityId)) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS134)